parser-service/src/main/java/org/mule/parser/service/strategy/WithFallbackParsingStrategy.java

```diff
@@ -6,35 +6,68 @@
+import org.mule.parser.service.result.DefaultParseResult;
 import org.mule.parser.service.result.DefaultParsingIssue;
 import org.mule.parser.service.result.ParseResult;
 import org.mule.parser.service.result.ParsingIssue;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import com.google.common.collect.ImmutableList;
 
 public class WithFallbackParsingStrategy implements ParsingStrategy {
   private static final AMFParsingStrategy AMF_DELEGATE = new AMFParsingStrategy();
+  public static final String RAML_FORMAT = "RAML";
   private ScheduledExecutorService executor;
 
   @Override
   public ParseResult parse(ApiReference ref) {
-    ParseResult amfResult = AMF_DELEGATE.parse(ref);
-    if (amfResult.success()) {
-      return amfResult;
+    ParseResult parseResult = AMF_DELEGATE.parse(ref);
+    if (!parseResult.success() && StringUtils.equals(ref.getFormat(), RAML_FORMAT)){
+      ParseResult ramlResult = new RamlParsingStrategy(addExecutor(parseResult)).parse(ref);
+      List<ParsingIssue> errors = joinErrors(parseResult.getErrors(), ramlResult.getErrors());
+      List<ParsingIssue> warnings = joinErrors(parseResult.getWarnings(), ramlResult.getWarnings());
+      DefaultParseResult delegate = createDelegate(ramlResult, errors, warnings);
+      parseResult = new FallbackParseResult(delegate);
     }
-    ReferencesResolver referencesResolver = new ReferencesResolver(amfResult);
+    return parseResult;
+  }
 
-    if(executor != null){
-      referencesResolver.setExecutor(executor);
+  private DefaultParseResult createDelegate(ParseResult ramlResult, List<ParsingIssue> sourceErrors, List<ParsingIssue> sourceWarnings) {
+    List<ParsingIssue> errors = new ArrayList<>();
+    List<ParsingIssue> warnings = new ArrayList<>();
+    if(!ramlResult.success()){
+      errors = sourceErrors;
+      warnings = sourceWarnings;
     }
-    ParseResult ramlResult = new RamlParsingStrategy(referencesResolver).parse(ref);
-    return new FallbackParseResult(ramlResult);
+    DefaultParseResult defaultParseResult = new DefaultParseResult(ramlResult.get(), errors, warnings);
+    return defaultParseResult;
+  }
+
+  private ReferencesResolver addExecutor(ParseResult amfResult) {
+      ReferencesResolver referencesResolver = new ReferencesResolver(amfResult);
+      if(executor != null){
+        referencesResolver.setExecutor(executor);
+      }
+    return referencesResolver;
+  }
+
+  private List<ParsingIssue> joinErrors(List<ParsingIssue> amfIssues, List<ParsingIssue> ramlIssues) {
+    List<ParsingIssue> amfIssuesWithTitle = new ArrayList<>();
+    List<ParsingIssue> ramlIssuesWithTitle = new ArrayList<>();
+    amfIssues.forEach(parsingIssue ->{amfIssuesWithTitle.add(new DefaultParsingIssue("\nAMF: " + parsingIssue.cause()));});
+    ramlIssues.forEach(parsingIssue ->{ramlIssuesWithTitle.add(new DefaultParsingIssue("\nRAML: " + parsingIssue.cause()));});
```

Comment:
Is the line break needed? I would leave the job to the logger or formatter.

Reply:
Will extract to constants, this is for the case of AMF and RAML parsing errors, to know where it comes from.